dagshub/streaming/filesystem.py

```diff
@@ -147,15 +127,17 @@ def __init__(self,
         self.token = token or config.token
         self.timeout = timeout or config.http_timeout
 
-        response = self._api_listdir('')
-        if response.status_code < 400:
-            pass
-        else:
+        self._listdir_cache: Dict[Tuple[str, bool], Optional[List[ContentAPIEntry]]] = {}
```

Comment:
Yeaaa, this was just me rushing to rewrite the caching because it was a global singleton, and that was how it was first implemented.
I think your structure makes more sense and is less magical